xbmc/pvr/channels/PVRChannelGroupInternal.cpp

```diff
@@ -285,24 +286,38 @@ std::vector<std::shared_ptr<CPVRChannel>> CPVRChannelGroupInternal::RemoveDelete
   bool channelsDeleted = false;
 
   const std::shared_ptr<CPVRDatabase> database = CServiceBroker::GetPVRManager().GetTVDatabase();
-  if (!database)
+  const auto epgDatabase = CServiceBroker::GetPVRManager().EpgContainer().GetEpgDatabase();
```

Comment:
Could you do me a favor and not use auto here. I prefer to use auto only for more complex types than shared pointers.

Reply:
Done